mongo/change_stream.go

```diff
@@ -395,7 +395,16 @@ func (cs *ChangeStream) createPipelineOptionsDoc() bsoncore.Document {
 	}
 
 	if cs.options.FullDocument != nil {
-		plDoc = bsoncore.AppendStringElement(plDoc, "fullDocument", string(*cs.options.FullDocument))
+		// Only append a default "fullDocument" field if wire version is less than 6 (3.6). Otherwise,
```

Comment:
We never correctly implemented GODRIVER-2329. In particular, we were sending `fullDocument: "default"` in the change stream creation aggregation pipeline when it really wasn't necessary on 3.6+ (the server assumes you want the default behavior when it's not specified). This meant that the observed operation of `createChangeStream` in the unified test runner contained `fullDocument: "default"` if `fullDocument` was not specified. Expected events like [this one](https://github.com/mongodb/mongo-go-driver/pull/917/files#diff-0ed12ca33e2369d7671566ebb7407cd385045e6b9348c214afc7017e689d2d7fR193) saw `fullDocument: "default"` as an extra key and would error [here](https://github.com/mongodb/mongo-go-driver/blob/master/mongo/integration/unified/matches.go#L106).